integrations/weaviate/src/haystack_integrations/document_stores/weaviate/auth.py

```diff
@@ -0,0 +1,170 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field, fields
+from typing import Any, Dict, Type
+
+from haystack.core.errors import DeserializationError
+from haystack.core.serialization import default_from_dict, default_to_dict
+from haystack.utils.auth import Secret, deserialize_secrets_inplace
+
+from weaviate.auth import AuthApiKey as WeaviateAuthApiKey
+from weaviate.auth import AuthBearerToken as WeaviateAuthBearerToken
+from weaviate.auth import AuthClientCredentials as WeaviateAuthClientCredentials
+from weaviate.auth import AuthClientPassword as WeaviateAuthClientPassword
+
+
+@dataclass
+class AuthCredentials(ABC):
+    """
+    Base class for all auth credentials supported by WeaviateDocumentStore.
+    Can be used to deserialize from dict any of the supported auth credentials.
+    """
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Converts the object to a dictionary representation for serialization.
+        """
+        # We assume all fields are Secret instances
```

Comment:
We should check instead of assuming, especially if the `expires_in` field is going to be an `int` (see below.)